src/network/serverpackethandler.cpp

```diff
@@ -1154,8 +1154,11 @@ void Server::handleCommand_Interact(NetworkPacket *pkt)
 			u16 src_original_hp = pointed_object->getHP();
 			u16 dst_origin_hp = playersao->getHP();
 
-			pointed_object->punch(dir, &toolcap, playersao,
+			s16 wear = pointed_object->punch(dir, &toolcap, playersao,
 					time_from_last_punch);
+			bool result = punchitem.addWear(wear, m_itemdef);
+			if (result && playersao->setWieldedItem(punchitem))
+				SendInventory(playersao, true);
```

Comment:
Inventory sending happens automatically each step, triggered by `setWieldedItem`. I don't think prioritizing this packet is needed here, unless it's an empty itemstack (broke).